src/Strategy/Node/Pnpm/PnpmLock.hs

```diff
@@ -283,29 +318,31 @@ buildGraph lockFile = withoutLocalPackages $
     -- >      events: 1.1.1
     -- >    dev: false
     -- @
-    for_ (toList $ packages lockFile) $ \(pkgKey, pkgMeta) -> do
+    for_ (toList lockFile.packages) $ \(pkgKey, pkgMeta) -> do
       let deepDependencies =
-            toList (dependencies pkgMeta)
-              <> toList (peerDependencies pkgMeta)
+            Map.toList (dependencies pkgMeta)
+              <> Map.toList (peerDependencies pkgMeta)
+              <> fromMaybe mempty (Map.lookup pkgKey lockFile.lockFileSnapshots.snapshots)
 
       let (depName, depVersion) = case getPkgNameVersion pkgKey of
             Nothing -> (pkgKey, Nothing)
             Just (name, version) -> (name, Just version)
-      let parentDep = toDependency depName depVersion pkgMeta
+      let parentDep = toDependency False depName depVersion pkgMeta
 
       -- It is ok, if this dependency was already graphed as direct
       -- @direct 1 <> deep 1 = direct 1@
       deep parentDep
 
       for_ deepDependencies $ \(deepName, deepVersion) -> do
-        maybe (pure ()) (edge parentDep) (toResolvedDependency deepName deepVersion)
+        maybe (pure ()) (edge parentDep) (toResolvedDependency False deepName deepVersion)
   where
     getPkgNameVersion :: Text -> Maybe (Text, Text)
     getPkgNameVersion = case lockFileVersion lockFile of
       PnpmLock4Or5 -> getPkgNameVersionV5
       PnpmLock6 -> getPkgNameVersionV6
       PnpmLockLt4 _ -> getPkgNameVersionV5 -- v3 or below are deprecated and are not used in practice, fallback to closest
-      PnpmLockGt6 _ -> getPkgNameVersionV6 -- at the time of writing there is no v7, so default to closest
+      PnpmLockV678 _ -> getPkgNameVersionV6 -- at the time of writing there is no v7, so default to closest
```

Comment:
Do we still need `PnpmLock6` in this enum, or should it be merged into `PnpmLockV678`?

I'm finding it confusing that the `6` is included in `PnpmLockV678`, and it seems like `PnpmLockV678` is the same as `PnpmLock6` in behaviour